contents/tutorials/react-charts.md

```diff
@@ -0,0 +1,451 @@
+---
+title: How to use React Charts to visualize analytics data (with examples)
+date: 2024-09-12
+author:
+ - ian-vanagas
+tags:
+ - product analytics
+---
+
+[React Charts](https://react-charts.tanstack.com/) is a popular visualization and charting library for React. It provides a simplified set of performant charts to use with analytics data from PostHog.
+
+To provide examples of what you can do with it, we create a basic Next.js app, set it up to fetch data from [PostHog's query API](/docs/api/query), and visualize it with React Charts.
+
+> **Don't want to do all the work of querying and visualizing?** You can always share or embed insights, dashboards, and more. See [sharing docs](/docs/product-analytics/sharing) for more.
+
+## Creating a Next.js app
+
+To create a Next.js app, run the following command, choose all the default options, and select the `app` router.
+
+```bash
+npx create-next-app@latest react-charts-example
+```
+
+This creates a `react-charts-example` folder with everything we need to get started.
+
+## Querying data from PostHog
+
+Assuming you have data in PostHog to query, the next step is to set up our query request to PostHog.
+
+> **Don't have data to query?** Check out [our guide to setting up analytics in Next.js](/tutorials/nextjs-app-directory-analytics) to start capturing some.
+
+Start by creating a personal API key. You can do this by going to [personal API keys](https://us.posthog.com/settings/user-api-keys) in your project settings, clicking **Create personal API key**, giving it a label, choosing the **Performing analytics queries** scope (AKA query read), and clicking **Create key**.
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/light_api_468581fdca.png" 
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/dark_api_a755bfb6ac.png"
+  alt="Creating a personal API key in PostHog" 
+  classes="rounded"
+/>
+
+> **⚠️ Warning:** The following is a simplified example. It will expose your private PostHog data to the public internet, so don't do this in real life. You want to make sure your personal API key isn't exposed and your query request happens securely on the server side.
```

Comment:
This warning makes it seem like by creating an API key, all of our posthog data automatically becomes public. It also doesnt make sense here, since we havent yet added the code for the apikey on the client.

Edit: I dont actually see the part where we expose the API key? It's in the /api folder, which is server side, not client side? So we dont need this warning at all?

Reply:
If you implemented it this way, anyone could query the `api/query` endpoint with whatever HogQL payload they want and get whatever data they want. 

The right way to do it is to have the payload in the server side as well, but that would require changes to two files on every example which I didn't want to do for brevity.

Reply:
rewrote to clarify again